# Variant-Caling-in-RNA-Seq-
![image](https://media.springernature.com/lw685/springer-static/image/chp%3A10.1007%2F978-1-0716-2293-3_13/MediaObjects/504146_1_En_13_Fig1_HTML.png)
# Variant-Calling-from-RNA-seq-Data-Using-the-GATK
<img align="right" alt="coding" width ="800" src= "https://camo.githubusercontent.com/8dd320d95014aec88352fc2dd35a837cf5614910bd2f7d5f3a44a73207d6de55/68747470733a2f2f75732e762d63646e2e6e65742f353031393739362f75706c6f6164732f46696c6555706c6f61642f66612f6536306563663839626431623236343564396663653638636366333931392e706e67">

>In brief, the key modifications made to the DNAseq Best Practices focus on handling splice junctions correctly, which involves specific mapping and pre-processing procedures, as well as some new functionality in the HaplotypeCaller. Here is a detailed overview:
<img align= "right" alt="coding" width="800" src="https://camo.githubusercontent.com/032092918e223dd9f9e6e787e103e1a435bdbf0edf6cc67b1bd20ebd2aa94234/68747470733a2f2f75732e762d63646e2e6e65742f353031393739362f75706c6f6164732f46696c6555706c6f61642f63392f6163343637383462653339663331666139373662356163393434646531372e706e67">

# Workflow
